goodmap/formatter.py

```diff
@@ -9,10 +10,22 @@ def safe_gettext(lista):
 
 
 def prepare_pin(place, visible_fields):
+    type_of_place = get_type_of_place(place)
     pin_data = {
         "title": place["name"],
-        "subtitle": lazy_gettext(place["type_of_place"]),
+        "subtitle": lazy_gettext(type_of_place),
         "position": place["position"],
-        "data": {gettext(field): safe_gettext(place[field]) for field in visible_fields}
+        "data": {
+            gettext(field): safe_gettext(place[field]) for field in visible_fields
+        },
     }
     return pin_data
+
+
+def get_type_of_place(place):
+    # TODO: should we require 'type_of_place'? For now, this workaround
+    try:
+        return place["type_of_place"]
+    except KeyError:
+        logging.error("Missing key 'type_of_place'")
```

Comment:
Isn't original error (without logging) exactly the same?

Reply:
No, there is no stack trace in this solution. I think we should either stop the application from starting if it lacks proper data, or accept this situation and stop the framework from printing stack trace.

Reply:
I am all for no 1. Some db checker would be great.